source/Ship.cpp

```diff
@@ -3374,6 +3375,66 @@ double Ship::IdleHeat() const
 	double dissipation = HeatDissipation() + activeCooling / MaximumHeat();
 	if(!dissipation) return production ? numeric_limits<double>::max() : 0;
 	return production / dissipation;
+	*/
+
+	// The Secant Method, closely related to Newton's Method,
+	// is a robust approximation, approaching the zero of any curve
+	// quickly and cheaply. It does not provide infinite accuracy,
+	// but in exchange it is trivial to add more functionality to
+	// the curve it is approximating a zero of.
+	// To do so- implement your heat change in ship::DoGeneration()
+	// and then add the same change to Ship::NetIdleHeatAt().
+
+	int attempts = 10;
+	double firstGuess = 0.;
+	double firstOutput = NetIdleHeatAt(firstGuess);
+
+	double secondGuess = MaximumHeat();
+	double secondOutput = NetIdleHeatAt(secondGuess);
+
+	double middlingGuess;
+	double middlingOutput;
+
+	// Guard clause, to catch weird cases.
+	if(firstGuess == secondGuess || firstOutput <= secondOutput)
+		return secondOutput > 0. ? numeric_limits<double>::max() : 0;
```

Comment:
Just a question, why exactly are you returning the maximum double here?

Reply:
in the case where the maximum double is returned, heat level is a positive feedback loop (rather than a negative feedback loop) and if your heat is ever positive it'll explode out towards positive infinity. Thus, our idle heat is the maximum double.

Reply:
ok, thanks, you can resolve this now